### DISCUSSION TITLE
VRCSubs - Chatbubble OSC Speech to Text

### DISCUSSION THREAD

**Discussion by cyberkitsune** [vrchat-community/osc#135] [Show and tell]
### VRCSubs - https://github.com/cyberkitsune/vrc-osc-scripts
Using the Python `SpeechRecognition` package, I made a small OSC helper script to transcribe your speech into text as you talk, and send VRChat Chatbox messages with what you said in _almost_ real-time! It currently utilizes the Google Speech Search API, with the API key provided by the `SpeechRecognition` package. (While free, it does make it less ideal for a real-time transcriber, though...)

You can find it in my [vrc-osc-scripts](https://github.com/cyberkitsune/vrc-osc-scripts) repo under the `VRCSubs/` folder!

### Demonstration
![VRCSubs in action!](https://raw.githubusercontent.com/cyberkitsune/vrc-osc-scripts/main/img/subtitles.gif)

**Comment by momo-the-monster** [comment]
@cyberkitsune - thanks for sharing! Your tools have been added to [the official listing](https://github.com/vrchat-community/osc/wiki/Resources#misc). Since both of the tools you submitted are in the same repository, they've been combined into one listing there.
